src/auth/auth.controller.ts

```diff
@@ -0,0 +1,49 @@
+import {
+  Body,
+  Controller,
+  HttpCode,
+  HttpStatus,
+  Post,
+  UseGuards,
+} from '@nestjs/common';
+import { AuthService } from './auth.service';
+import { CreateUserDto } from './input/auth.dto';
+import { Tokens } from './models/tokens.type';
+import { GetCurrentUser, GetCurrentUserId, Public } from '../common/decorators';
+import { RtGuard } from '../common/guards';
+
+@Controller('auth')
+export class AuthController {
+  constructor(private authService: AuthService) {}
+
+  @Public()
+  @Post('register')
+  @HttpCode(HttpStatus.CREATED)
+  register(@Body() dto: CreateUserDto): Promise<Tokens> {
+    return this.authService.register(dto);
+  }
```

Comment:
I'm not sure if we should return access tokens on registration. There are apps that do it, but if we look at our task requirements we should do this only after user confirms the email address. So I think we should just return 200 OK here.